.github/workflows/go.yml

```diff
@@ -0,0 +1,25 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
```

Comment:
We should test latest golang version as well. Let's make a matrix with 1.17, 1.18 and 1.19.

Reply:
i have added for other OS support as well, do let me if you have different thought on this.